samplePDF/samplePDFFDBase.cpp

```diff
@@ -1414,11 +1333,118 @@ void samplePDFFDBase::addData(TH2D* Data) {
   }
 }
 
+void samplePDFFDBase::SetupNuOscillator() {
+  OscillatorFactory* OscillFactory = new OscillatorFactory();
+  
+  NuOscProbCalcers = std::vector<OscillatorBase*>((int)MCSamples.size());
+  for (int iSample=0;iSample<(int)MCSamples.size();iSample++) {
+    MACH3LOG_INFO("Setting up NuOscillator::Oscillator object in OscillationChannel: {}/{}", iSample, MCSamples.size());
+    NuOscProbCalcers[iSample] = OscillFactory->CreateOscillator(NuOscillatorConfigFile);
+    
+    if (!NuOscProbCalcers[iSample]->EvalPointsSetInConstructor()) {
+      std::vector<_float_> EnergyArray;
+      for (int iEvent=0;iEvent<(int)MCSamples[iSample].nEvents;iEvent++) {
+        //DB Remove NC events from the arrays which are handed to the NuOscillator objects
+        if (!MCSamples[iSample].isNC[iEvent]) {
+          EnergyArray.push_back(*(MCSamples[iSample].rw_etru[iEvent]));
+        }
+      }
+      std::sort(EnergyArray.begin(),EnergyArray.end());
+      
+      NuOscProbCalcers[iSample]->SetEnergyArrayInCalcer(EnergyArray);
+
+      //============================================================================
+      //DB Atmospheric only part
+      if (MCSamples[iSample].rw_truecz != NULL) { //Can only happen if truecz has been initialised within the experiment specific code
+	std::vector<_float_> CosineZArray;
+	for (int iEvent=0;iEvent<(int)MCSamples[iSample].nEvents;iEvent++) {
+	  //DB Remove NC events from the arrays which are handed to the NuOscillator objects
+	  if (!MCSamples[iSample].isNC[iEvent]) {
+	    CosineZArray.push_back(*(MCSamples[iSample].rw_truecz[iEvent]));
+	  }
+	}
+	std::sort(CosineZArray.begin(),CosineZArray.end());
+	
+	NuOscProbCalcers[iSample]->SetCosineZArrayInCalcer(CosineZArray);
+      }
+      //============================================================================
+      
+    }
+    
+    NuOscProbCalcers[iSample]->Setup();
+
+    for (int iEvent=0;iEvent<(int)MCSamples[iSample].nEvents;iEvent++) {
+      // KS: Sry but if we use low memory we need to point to float not double...
+#ifdef _LOW_MEMORY_STRUCTS_
+      MCSamples[iSample].osc_w_pointer[iEvent] = &Unity_F;
```

Comment:
its better to just change how `Unity` is declared based on this define, rather than littering the code with hard to parse CPP directives